packages/client-core/src/admin/common/Project/ProjectDrawer.tsx

```diff
@@ -70,14 +70,14 @@ const ProjectDrawer = ({ open, inputProject, existingProject = false, onClose, c
 
   const projectUpdateStatus = useHookstate(getMutableState(ProjectUpdateState)[project.name]).value
 
-  const handleSubmit = async () => {
+  const handleSubmit = () => {
     try {
       if (existingProject && changeDestination) {
-        if (inputProject) await ProjectService.setRepositoryPath(inputProject.id, projectUpdateStatus.destinationURL)
```

Comment:
we need to still be awaiting all these calls, otherwise the UI will never reflect requests in progress

Reply:
The functions (in this case `setRepositoryPath`) has no await call in it, Is it still valid if it's defined as async function?

Reply:
The point is that these are API calls, they are necessarily going to be async.